pkgs/development/interpreters/python/hooks/editable-install-hook.sh

```diff
@@ -0,0 +1,30 @@
+editableInstallHook() {
+    echo "Executing editableInstallHook"
+    runHook preShellHook
+
+    # We make a temporary directory to hold the editable install. We don't
+    # really have a way to clean it up as it must exist past the shell lifetime
+    # for tools like direnv to work properly. However, the total contents are
+    # only a few kilobytes (independent of the size of the edited package) and
+    # the operating system will clean it up e.g. on reboot.
+    tmp_path=$(mktemp -d)
+    export PATH="$tmp_path/bin:$PATH"
+    export PYTHONPATH="$tmp_path/@pythonSitePackages@:$PYTHONPATH"
```

Comment:
Setting `$PYTHONPATH` has serious implications for dependencies leaking into other Python applications.

Reply:
This hook is only designed for use in a shell environment where you're running the Python application that's been set in `$PYTHONPATH`, so I'm not sure this is a huge concern.

Do you have suggestions on how to fix it? The old hook worked exactly like this too.

Reply:
> This hook is only designed for use in a shell environment where you're running the Python application that's been set in $PYTHONPATH

Which is something that you should never do anyway. `PYTHONPATH` should be seen as an escape hatch that is to be avoided as much as humanly possible.
The preferred workflow of using `buildEnv`/`withPackages` circumvents `PYTHONPATH` issues via wrappers.

> Do you have suggestions on how to fix it? The old hook worked exactly like this too.

I would rather just kill off `setuptoolsShellHook` entirely.
It's just bad.

Reply:
`PYTHONPATH` is already set up by default in a shell. I don't think adding one entry is going to make things worse.